compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -199,6 +201,25 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
             }
         }
     }
+    if !all_consts_ok {
```

Comment:
could we instead move the const evaluation earlier in this function so we wouldn't even build the basic blocks at all?

Reply:
Then we don't have `fx` yet so we'd have to duplicate `fx.eval_mir_constant`.

Reply:
urgh, right, I remember this.